domain/annotation/state/query.go

```diff
@@ -0,0 +1,179 @@
+// Copyright 2023 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package state
+
+import (
+	"fmt"
+
+	"github.com/canonical/sqlair"
+	"github.com/juju/errors"
+
+	"github.com/juju/juju/core/annotations"
+	annotationerrors "github.com/juju/juju/domain/annotation/errors"
+)
+
+// getAnnotationQueryForID provides a query for the given id, based on pre-computed queries for
+// GetAnnotations for different kinds of ids. We keep these static (avoiding dynamically generating
+// table names and fields) to keep things safe.
+func getAnnotationQueryForID(id annotations.ID) string {
+	var query string
+	switch id.Kind {
+	case annotations.KindModel:
+		query = `SELECT (key, value) AS (&Annotation.*) from annotation_model`
+	case annotations.KindMachine:
+		query = `
+SELECT (key, value) AS (&Annotation.*)
```

Comment:
Why not make the referential column just `uuid`, so the only variable is the name?

This and other places would DRY up nicely.